tasks/internal/update-fbc-catalog-task/tests/mocks.sh

```diff
@@ -65,23 +63,63 @@ function mock_build_progress() {
 
 function curl() {
   params="$*"
-  # Debug: always print the task name and curl params for troubleshooting
-  echo "DEBUG: TaskRun name: $(context.taskRun.name)" >&2
-  echo "DEBUG: curl params: $params" >&2
   
   if [[ "$params" =~ "--negotiate -u: https://pyxis.engineering.redhat.com/v1/repositories/registry/quay.io/repository/repo/image -o"* ]]; then
     tempfile="$5"
     echo -e '{ "fbc_opt_in": true }' > "$tempfile"
 
-  elif [[ "$params" =~ "-s https://fakeiib.host/builds?user=iib@kerberos&from_index=quay.io/scoheb/fbc-index-testing:"* ]]; then
+  elif [[ "$params" == *"-s https://fakeiib.host/builds"* ]] && [[ "$params" == *"from_index="* ]] && [[ "$params" == *"state="* ]] && [[ "$params" != *"/builds/1"* ]] && [[ "$params" != *"/api/v1/builds"* ]]; then
+    # Check params directly for registry-proxy pattern (most reliable)
+    is_registry_proxy_test=false
+    if [[ "$params" =~ registry-proxy.engineering.redhat.com/rh-osbs/iib-pub ]]; then
+      is_registry_proxy_test=true
+    fi
+    # Extract from_index from params to help with debugging
+    # Handle both URL-encoded and non-encoded versions
+    extracted_from_index=""
+    # Try multiple patterns to extract from_index
+    if [[ "$params" =~ from_index=([^&\ ]+) ]]; then
+      extracted_from_index="${BASH_REMATCH[1]}"
+    elif [[ "$params" =~ from_index%3D([^&\ ]+) ]]; then
+      # URL-encoded equals sign
+      extracted_from_index="${BASH_REMATCH[1]}"
+    fi
+    # URL decode if needed (simple case for common characters)
+    if [[ -n "$extracted_from_index" ]]; then
```

Comment:
Can you add a comment explaining what strings you are expecting to read/get/extract just to help out the readability?